src/sentry/tasks/assemble.py

```diff
@@ -251,53 +251,68 @@ def _normalize_debug_id(debug_id: Optional[str]) -> Optional[str]:
         return None
 
 
-def _extract_debug_ids_from_manifest(manifest: dict) -> List[Tuple[SourceFileType, str]]:
+def _extract_debug_ids_from_manifest(
+    manifest: dict,
+) -> Tuple[Optional[str], List[Tuple[SourceFileType, str]]]:
     debug_ids_with_types = []
 
+    # We also want to extract the bundle_id which is also known as the bundle debug_id. This id is used to uniquely
+    # identify a specific ArtifactBundle in case for example of future deletion.
+    #
+    # If no id is found, it means that we must have an associated release to this ArtifactBundle, through the
+    # ReleaseArtifactBundle table.
+    bundle_id = manifest.get("debug_id")
+    if bundle_id is not None:
+        bundle_id = _normalize_debug_id(bundle_id)
+
     files = manifest.get("files", {})
     for file_path, info in files.items():
         headers = _normalize_headers(info.get("headers", {}))
-        if (debug_id := headers.get("debug-id", None)) is not None:
+        if (debug_id := headers.get("debug-id")) is not None:
```

Comment:
Is there a difference in python between `headers.get("debug-id", None) is not None:` and ``headers.get("debug-id") is not None:`? I've seen we provide default value for getters, and then use `is/is not` immediately after this in a few places (eg. in `allow_scraping` in processor)

Reply:
That was my thinking, and that's why I did not understand why we had `get(key, def) is not def` in few places :P